### PR TITLE
Public index page

### PR DESCRIPTION
**Pull request by juliajulyy**
**STATUS**
READY/IN DEVELOPMENT

**DESCRIPTION**

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [ ] "let!" is used in specs

### CODE REVIEW

File: app/views/layouts/application.html.haml [hunk 1]
```diff
@@ -8,4 +8,12 @@
     = stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload'
     = javascript_include_tag 'application', 'data-turbolinks-track': 'reload'
   %body
-    = yield
+    = render 'shared/header'
+
+    %main{:class => "content #{yield :index} container"}
+
```
Comment (BogdanBusko) [inline app/views/layouts/application.html.haml]: delete enter

File: app/assets/stylesheets/public_pages.scss [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  min-height: calc(100vh - #{$footer-size});
+  z-index: 1;
+  min-width: 100vw;
+  margin: 0;
+  animation: slide 20s linear infinite;
+  text-align: center;
+  background-image: url("home-page-back.png");
+}
+
+
```
Comment (BogdanBusko) [inline app/assets/stylesheets/public_pages.scss]: delete enter

File: app/views/public_pages/index.html.haml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+- content_for :index do
+  animated
```
Comment (BogdanBusko) [inline app/views/public_pages/index.html.haml]: write it in one line

**Comment by juliajulyy** [comment]
image

File: app/assets/stylesheets/public_pages.scss [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    text-decoration: none;
+    color: #f97300;
+    border-radius: 47%;
+    background-color: #fff;
+  }
+}
+
+/*footer*/
+
+.footer {
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/public_pages.scss]: split this file into smaller

header.scss
footer.scss

File: app/assets/stylesheets/public_pages.scss [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+      color: #f97300;
+      background-color: white;
+      transition: .5s ease-in-out ;
+    }
+  }
+}
+
+/*background-animation*/
+
+.animated {
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/public_pages.scss]: animations.scss

File: app/views/public_pages/index.html.haml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+- content_for :index do 'animated' end
```
Comment (igorkasyanchuk) [inline app/views/public_pages/index.html.haml]: good, but change name from "index" to something like "container_class"

File: bin/rails [hunk 7]
```diff
@@ -1,4 +1,10 @@
 #!/usr/bin/env ruby
+if ENV['RAILS_ENV'] == 'test'
+  require 'simplecov'
+  SimpleCov.start 'rails'
+  puts "required simplecov"
```
Comment (igorkasyanchuk) [inline bin/rails]: remove this puts

File: config/routes.rb [hunk 8]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-end
+
+  root 'public_pages#index'
```
Comment (igorkasyanchuk) [inline config/routes.rb]: rename controller from "public_pages" to "home"

File: app/assets/stylesheets/footer.scss [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  width: 100%;
+  padding-right: 15px;
+  padding-left: 15px;
+  margin-right: auto;
+  margin-left: auto;
+  color: white;
+  font-size: .8rem;
+}
+
+.list-item {
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/footer.scss]: put all CSS inside .footer { ... }

since it belongs to it

File: app/assets/stylesheets/header.scss [hunk 10]
```diff
@@ -0,0 +1,73 @@
+a:hover {
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/header.scss]: why a:hover is here?

create "base.scss" and put general rules there

File: app/assets/stylesheets/home.scss [hunk 11]
```diff
@@ -0,0 +1,55 @@
+$footer-size: 50px;
+$navbar-color: darkblue;
+
+html, body {
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/home.scss]: move some CSS into base.scss
which is not related to home page

File: config/routes.rb [hunk 12]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-end
+
+  get 'home/index'
```
Comment (igorkasyanchuk) [inline config/routes.rb]: remove this "home/index" from here

File: spec/controllers/home_controller_spec.rb [hunk 13]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe HomeController, type: :controller do
+
```
Comment (igorkasyanchuk) [inline spec/controllers/home_controller_spec.rb]: додай ще таке як 

render_views

має бути в тестах rspec for controllers

File: Gemfile [hunk 14]
```diff
@@ -3,65 +3,42 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
...
+gem 'coffee-rails',     '~> 4.2'
+gem 'turbolinks',       '~> 5'
+gem 'jbuilder',         '~> 2.5'
+gem 'bootsnap',         '>= 1.1.0', require: false
 
 group :development, :test do
-  gem 'rspec-rails', '~> 3.8'
-  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
+  gem 'rspec-rails',    '~> 3.8'
+  gem 'shoulda-matchers', '~> 3.1', '>= 3.1.1'
```
Comment (romaslmd) [inline Gemfile]: u dont really need this in dev env

File: bin/rails [hunk 15]
```diff
@@ -1,4 +1,9 @@
 #!/usr/bin/env ruby
+if ENV['RAILS_ENV'] == 'test'
```
Comment (romaslmd) [inline bin/rails]: is this really necessary?

Comment (juliajulyy) [reply bin/rails]: You're right and it's really doesn't important

**Pull request merged**